code/modules/mob/living/living_defense.dm

```diff
@@ -65,60 +65,46 @@
-/mob/living/hitby(atom/movable/AM as mob|obj,var/speed = 5)//Standardization and logging -Sieve
+/mob/living/hitby(atom/movable/AM, skipcatch, hitpush = 1, blocked = 0)//Standardization and logging -Sieve
 	if(istype(AM, /obj/item))
 		var/obj/item/I = AM
 		var/zone = ran_zone("chest", 65)//Hits a random part of the body, geared towards the chest
-		var/dtype = BRUTE
+		var/dtype = BRUTE
+		var/volume = I.get_volume_by_throwforce_and_or_w_class()
 		if(istype(I, /obj/item/weapon))
 			var/obj/item/weapon/W = I
 			dtype = W.damtype
-			if(W.hitsound && W.throwforce > 0)
-				playsound(loc, W.hitsound, 30, 1, -1)
-
-		//run to-hit check here
-
-		var/throw_damage = I.throwforce*(speed/5)
-
-		src.visible_message("\red [src] has been hit by [I].")
-		var/armor = run_armor_check(zone, "melee", "Your armor has protected your [parse_zone(zone)].", "Your armor has softened hit to your [parse_zone(zone)].", I.armour_penetration)
-
-		apply_damage(throw_damage, dtype, zone, armor, is_sharp(I), has_edge(I), I)
-
-		I.throwing = 0		//it hit, so stop moving
-
-		if(ismob(I.thrower))
-			var/mob/M = I.thrower
-			if(M)
-				create_attack_log("<font color='orange'>Has been hit with a [I], thrown by [key_name(M)]</font>")
-				M.create_attack_log("<font color='red'>Hit [key_name(src)] with a thrown [I]</font>")
-				if(!istype(src,/mob/living/simple_animal/mouse))
-					msg_admin_attack("[key_name_admin(src)] was hit by a [I], thrown by [key_name_admin(M)]")
-
-		// Begin BS12 momentum-transfer code.
-		if(I.throw_source && speed >= 15)
-			var/obj/item/weapon/W = I
-			var/momentum = speed/2
-			var/dir = get_dir(I.throw_source, src)
-
-			visible_message("\red [src] staggers under the impact!","\red You stagger under the impact!")
-			src.throw_at(get_edge_target_turf(src,dir),1,momentum)
-
-			if(!W || !src) return
-
-			if(W.sharp) //Projectile is suitable for pinning.
-				//Handles embedding for non-humans and simple_animals.
-				I.loc = src
-				embedded += I
-
-				var/turf/T = near_wall(dir,2)
-
-				if(T)
-					src.loc = T
-					visible_message("<span class='warning'>[src] is pinned to the wall by [I]!</span>","<span class='warning'>You are pinned to the wall by [I]!</span>")
-					src.anchored = 1
-					src.pinned += I
+			if(W.throwforce > 0) //If the weapon's throwforce is greater than zero...
+				if(W.hitsound) //...and hitsound is defined...
+					playsound(loc, W.hitsound, 30, 1, -1) //...play the weapon's hitsound.
+				else //Otherwise, if hitsound isn't defined...
+					playsound(loc, 'sound/weapons/genhit1.ogg', volume, 1, -1) //...play genhit1.ogg.
+
+		else if(I.throwforce > 0) //Otherwise, if the item doesn't have a throwhitsound and has a throwforce greater than zero...
+			playsound(loc, 'sound/weapons/genhit1.ogg', volume, 1, -1)//...play genhit1.ogg
+		if(!I.throwforce)// Otherwise, if the item's throwforce is 0...
+			playsound(loc, 'sound/weapons/throwtap.ogg', 1, volume, -1)//...play throwtap.ogg.

```

Comment:
Is there anyway to keep that functionality? That would be great even as a percentage chance